test/runner/lib/cli.py

```diff
@@ -106,8 +113,13 @@ def main():
 
         try:
             args.func(config)
+            delegate_args = None
         except Delegate as ex:
-            delegate(config, ex.exclude, ex.require, ex.integration_targets)
+            # save delegation args for use once we exit the exception handler
+            delegate_args = (ex.exclude, ex.require, ex.integration_targets)
+
+        if delegate_args:
```

Comment:
This prevents Python 3 from using exception chaining when an unhandled exception is generated in `delegate()`. Since the `Delegate` exception is being used for control flow instead of error reporting, there's little value in including it in the traceback if `delegate()` fails.